src/BlockEntities/DispenserEntity.cpp

```diff
@@ -178,6 +189,64 @@ void cDispenserEntity::DropSpenseFromSlot(cChunk & a_Chunk, int a_SlotNum)
 			break;
 		}
 
+		case E_ITEM_BOTTLE_O_ENCHANTING:
+		{
+			if (SpawnProjectileFromDispenser(BlockX, DispY, BlockZ, cProjectileEntity::pkExpBottle, GetShootVector(Meta) * 20 + Vector3d(0, 1, 0)) != cEntity::INVALID_ID)
+			{
+				m_Contents.ChangeSlotCount(a_SlotNum, -1);
+			}
+			break;
+		}
+
+		case E_ITEM_POTION:
+		{
+			if (SpawnProjectileFromDispenser(BlockX, DispY, BlockZ, cProjectileEntity::pkSplashPotion, GetShootVector(Meta) * 20 + Vector3d(0, 1, 0), &SlotItem) != cEntity::INVALID_ID)
+			{
+				m_Contents.ChangeSlotCount(a_SlotNum, -1);
+			}
+			break;
+		}
+
+		case E_ITEM_DYE:
+		{
+			if (SlotItem.m_ItemDamage != E_META_DYE_WHITE)
+			{
+				DropFromSlot(a_Chunk, a_SlotNum);
+				break;
+			}
+			if (m_World->GrowRipePlant(BlockX, DispY, BlockZ, true))
+			{
+				m_Contents.ChangeSlotCount(a_SlotNum, -1);
+			}
+			break;
```

Comment:
Ok, just checking. It feels wrong to me that one can dispense any dye other than bonemeal, but if Mojang decided that way...